cluster/gce/coreos/node.yaml

```diff
@@ -113,48 +50,62 @@ coreos:
+
 
-    - name: kubernetes-preparation.service
+    - name: kubernetes-install-node.service
       command: start
       content: |
         [Unit]
-        Description=Configure Node For Kubernetes service
-        Requires=kubernetes-install-minion.service
-        After=kubernetes-install-minion.service
-        Requires=kubernetes-install-rkt.service
-        After=kubernetes-install-rkt.service
+        Description=Install Kubernetes Server
+        Requires=network-online.target
+        After=network-online.target
+        Requires=kube-env.service
+        After=kube-env.service
         [Service]
         Type=oneshot
         RemainAfterExit=yes
         EnvironmentFile=/etc/kube-env
-        # TODO(dawnchen): Push this to separate write-files
-        ExecStart=/run/configure-hostname.sh
+        ExecStartPre=/usr/bin/mkdir -p /opt/kubernetes/pkg
+        ExecStartPre=/usr/bin/curl --location --create-dirs --output /opt/kubernetes/pkg/kubernetes-server-linux-amd64.tar.gz ${SERVER_BINARY_TAR_URL}
+        ExecStart=/usr/bin/tar xf /opt/kubernetes/pkg/kubernetes-server-linux-amd64.tar.gz -C /opt --overwrite
 
     - name: kubelet.service
       command: start
       content: |
         [Unit]
         Description=Run Kubelet service
-        Requires=kubernetes-preparation.service
-        After=kubernetes-preparation.service
+        Requires=network-online.target
+        After=network-online.target
+        Requires=kube-env.service
+        After=kube-env.service
+        Requires=kubernetes-download-templates.service
+        After=kubernetes-download-templates.service
         [Service]
         EnvironmentFile=/etc/kube-env
-        ExecStartPre=/usr/bin/mkdir -p /etc/kubernetes/manifests
-        ExecStartPre=/run/setup-auth.sh
+        ExecStartPre=/usr/bin/curl --fail --silent --show-error \
+        -H "X-Google-Metadata-Request: True" \
+        -o /run/configure-kubelet.sh \
+        http://metadata.google.internal/computeMetadata/v1/instance/attributes/configure-kubelet
+        ExecStartPre=/usr/bin/chmod 0755 /run/configure-kubelet.sh
+        ExecStartPre=/run/configure-kubelet.sh
         ExecStart=/opt/kubernetes/server/bin/kubelet \
-        --api-servers=https://${KUBERNETES_MASTER_NAME}.c.${PROJECT_ID}.internal \
+        --api-servers=https://${INSTANCE_PREFIX}-master \
+        --enable-debugging-handlers=true \
+        --cloud-provider=gce \
         --config=/etc/kubernetes/manifests \
-        --allow-privileged=False \
+        --allow-privileged=true \
         --v=2 \
-        --cluster-dns=10.0.0.10 \
-        --cluster-domain=cluster.local \
+        --cluster-dns=${DNS_SERVER_IP} \
+        --cluster-domain=${DNS_DOMAIN} \
         --logtostderr=true \
         --container-runtime=${KUBERNETES_CONTAINER_RUNTIME} \
         --rkt-path=/opt/rkt/rkt/rkt \
-        --configure-cbr0=${KUBERNETES_CONFIGURE_CBR0}
+        --configure-cbr0=${KUBERNETES_CONFIGURE_CBR0} \
+        --pod-cidr=${MASTER_IP_RANGE} \
```

Comment:
I believe this is unnecessary (and more or less a no-op) because --reconcile-cidr defaults to true. So it will ask apiserver what podIP range it should use / configure bridge / kick docker (or I'm missing something). But it does seem odd to use the same podIP range for multiple nodes.